### PR TITLE
i#2812 sigaction: support disabled vsyscall32

### PR DESCRIPTION
**Pull request by alpire**
This ensures that SA_RESTORER is passed to sigaction on 32-bit builds. A signal
restorer is required when vsyscall32 is disabled, which removes the default
signal restorer stub included in the vdso. Therefore, the application cannot
rely on the existence of a default signal handler, and provides one by setting
the SA_RESTORER flag.

Issue: #2812

### CODE REVIEW

File: core/unix/signal.c [hunk 1]
```diff
@@ -276,8 +276,11 @@ is_sys_kill(dcontext_t *dcontext, byte *pc, byte *xsp, siginfo_t *info);
 int
 sigaction_syscall(int sig, kernel_sigaction_t *act, kernel_sigaction_t *oact)
 {
-#if defined(X64) && !defined(VMX86_SERVER) && defined(LINUX)
+#if !defined(VMX86_SERVER) && defined(LINUX)
     /* PR 305020: must have SA_RESTORER for x64 */
+    /* Must have SA_RESTORER to handle vsyscall32 being disabled
+       (see https://github.com/DynamoRIO/dynamorio/issues/2812)
```
Comment (derekbruening) [inline core/unix/signal.c]: style nit: missing leading '*'

File: core/unix/signal.c [hunk 2]
```diff
@@ -1312,8 +1315,11 @@ set_handler_sigact(kernel_sigaction_t *act, int sig, handler_t handler)
      */
     act->flags |= SA_RESTART;
 
-#if defined(X64) && !defined(VMX86_SERVER) && defined(LINUX)
+#if !defined(VMX86_SERVER) && defined(LINUX)
     /* PR 305020: must have SA_RESTORER for x64 */
+    /* Must have SA_RESTORER to handle vsyscall32 being disabled
+       (see https://github.com/DynamoRIO/dynamorio/issues/2812)
```
Comment (derekbruening) [inline core/unix/signal.c]: Same 2 comments here

File: core/unix/signal.c (hunk 1 above)
Comment (derekbruening) [inline core/unix/signal.c]: We use "i#2812" rather than the full issue URL

**Comment by derekbruening** [comment]
Thank you for the patch!  I just had a couple of minor style requests.

**Comment by alpire** [comment]
Thanks for the comments. I just pushed a new commit to fix the style issues.

**Comment by fhahn** [comment]
run aarch64 tests

**Comment by derekbruening** [comment]
What happened to Travis?  Did something change with the webhooks?

**Comment by alpire** [comment]
Based on https://travis-ci.org/DynamoRIO/dynamorio/requests, it looks like Travis detected some abuse. I'm not quite sure why.

**Comment by derekbruening** [comment]
I reached out to Travis support and am waiting on a response.

**Comment by derekbruening** [comment]
Travis has resolved the issue.  Please push an empty commit or sthg to trigger a new build as I'm not seeing a way for me to trigger one in this case.

**Comment by derekbruening** [comment]
run aarch64 tests

**Pull request merged**